bigquery/tests/system.py

```diff
@@ -1609,6 +1609,31 @@ def test_nested_table_to_dataframe(self):
             row['record_col']['nested_record']['nested_nested_string'],
             'some deep insight')
 
+    def test_list_rows_page_size(self):
+        num_items = 7
+        page_size = 3
+        num_pages, num_last_page = divmod(num_items, page_size)
+
+        SF = bigquery.SchemaField
+        schema = [SF('string_col', 'STRING', mode='NULLABLE')]
+        to_insert = [{'string_col': 'item%d' % i} for i in range(num_items)]
+        table_id = 'test_table'
+        dataset = self.temp_dataset(_make_dataset_id('nested_df'))
+        table_arg = Table(dataset.table(table_id), schema=schema)
+        table = Config.CLIENT.create_table(table_arg)
+        self.to_delete.insert(0, table)
+        Config.CLIENT.insert_rows(table, to_insert, selected_fields=schema)
```

Comment:
Please don't use `insert_rows()` in this system test. This makes this test very flakey. Instead it's recommended to use `load_table_from_file()` (search for `BytesIO` in this file for an example)

Why? With `insert_rows()`, rows get inserted into the streaming buffer and only asynchronously get added to the real table. `list_rows()` does not return rows from the streaming buffer.